mingw-w64-git/start-ssh-agent.cmd

```diff
@@ -79,6 +79,14 @@
 @ENDLOCAL & @SET "SSH_AUTH_SOCK=%SSH_AUTH_SOCK%" ^
           & @SET "SSH_AGENT_PID=%SSH_AGENT_PID%"
 
+@REM Set variables globally
+@where setx
+@if %ERRORLEVEL% == 0 (
+	@echo Setup global vars
+	@setx SSH_AGENT_PID %SSH_AGENT_PID%
+	@setx SSH_AUTH_SOCK %SSH_AUTH_SOCK%
```

Comment:
As I mentioned [here](https://github.com/git-for-windows/git-sdk-64/pull/19#issuecomment-596419500), I don't think we can use `setx` by default: it would potentially conflict between independent MinGit/Git for Windows installations.

Reply:
I look on it as windows-user, not as linux-user, because this is Windows script.
1. ssh-agent is a "service" which continue to work after CMD-script finished
2. in result, when I see it in memory, I expect from it workable its functions in programs, like Git Extensions or any git-command in any new console

Currently, without my fix, we can see it, but it doesn't work. I think this is problem.
Looks like it was written for users, who run it in console and do not exit from this console. I think this is wrong for Windows users...

If you want limit surprises for user, kill the agent after user close the shell. (I don't know how to do it)
Because with current behavior I, as Windows user, was surprised and it was not easily to find the problem.
I think my solution (last executed agent - last is active) will bring less problems, than current (agent executed, but we can't use it).

Reply:
Well, I did mention that you can have _independent_ MinGit/Git for Windows installations. For example, if you run both Git CMD and GitHub Desktop, you have two separate instances you are using. In that case, the configuration you so carefully set in Git CMD will make things fail in GitHub Desktop (because you cannot share emulated Unix sockets, as `sshd` uses, between separate MSYS2 installations, which Git for Windows is a subset of).

Reply:
I just installed GitHub Desktop and it works fine without any conflicts with my global ssh-agent with keys.
How can I repeat the conflict? Can you describe the steps?

Reply:
GitHub Desktop _just_ updated their bundled Git version to v2.29.2, I believe. This uses [the same MSYS2 version](https://github.com/git-for-windows/build-extra/blob/HEAD/versions/package-versions-2.29.2.txt#L113) as [the current Git for Windows version](https://github.com/git-for-windows/build-extra/blob/3ebf95d6bfaf18805c8c681616955c1649fbc472/versions/package-versions-2.30.1.txt#L113).

If you install anything older than v2.29.1, e.g. v2.26.2, you should run into those conflicts.

This same situation would arise if we would update to a newer MSYS2 version in Git for Windows: GitHub Desktop traditionally uses the release train that is one behind the current bleeding edge of Git for Windows.

Now, it _might_ even work across certain MSYS2 runtime version boundaries. That's cool. But it's not a guarantee. And therefore I won't accept the `setx`-based version you proposed. And neither should you.